cloudinit/sources/DataSourceAliYun.py

```diff
@@ -29,6 +30,7 @@ def imdsv2_token_put_header(self):
 
     def __init__(self, sys_cfg, distro, paths):
         super(DataSourceAliYun, self).__init__(sys_cfg, distro, paths)
+        self.default_update_events = copy.deepcopy(self.default_update_events)
         self.default_update_events[EventScope.NETWORK].add(EventType.BOOT)
```

Comment:
This code was previously following the guidance in `cloudinit/sources/__init__.py`:
```
   226 ▎   # A datasource requiring to write network config on each system boot        
   227 ▎   # would call default_update_events['network'].add(EventType.BOOT).  
```

Since that comment will instruct other datasource writers to also write code that will break our tests, could we please update the comment in the base class to reflect current best practice?

Reply:
Done, thanks for the suggestion!